homeassistant/helpers/config_validation.py

```diff
@@ -376,9 +376,12 @@ def validator(value):
         """Validate ordered dict."""
         config = OrderedDict()
 
-        for key, val in value.items():
-            v_res = item_validator({key: val})
-            config.update(v_res)
+        try:
+            for key, val in value.items():
+                v_res = item_validator({key: val})
+                config.update(v_res)
+        except AttributeError:
```

Comment:
Rather check if it is a dict, like you do on line 416 below, the wide scope of this exception makes readability difficult